scripts/india_census/primary_religion_data/preprocess.py

```diff
@@ -0,0 +1,354 @@
+    """An object that represents Census Data and its variables.
+    
+    Attributes:
+        census_columns (list): It will have all the data column names of a dataset
+        census_year : Census year
+        csv_file_path : Path where cleaned csv file will be saved
+        data_file_path : Input XLS file from Census of India. Can be url or local path.
+        dataset_name : Census dataset name. Eg:Primary_Abstract_Data
+        existing_stat_var (list): List of existing stat vars that we don't need to generate
+        mcf (list): Description
+        mcf_file_path : Description
+        metadata_file_path : Description
+        raw_df : Raw census data as dataframe
+        stat_var_index (dict): local storage for census column name and corresponding statvar
+        tmcf_file_path : Path where generated tmcf file will be saved
+        data_categories: List data categories for which data is defined
+        data_category_column: Name of the data category column
+
+    """
+
+    def __init__(self, data_file_path, metadata_file_path, mcf_file_path,
+                 tmcf_file_path, csv_file_path, existing_stat_var, census_year,
+                 dataset_name, data_categories, data_category_column):
+        """
+        Constructor
+        
+        Args:
+            data_file_path :  Input XLS file from Census of India. Can be url or local path
+            metadata_file_path : Meta data csv file which has attribute details
+            mcf_file_path : Path where generated mcf file will be saved
+            tmcf_file_path : Path where generated tmcf file will be saved
+            csv_file_path : Path where cleaned csv file will be saved
+            existing_stat_var : List of existing stat vars that we don't need to generate
+            census_year : Census Year
+            dataset_name : Census dataset name. Eg:Primary_Abstract_Data
+            data_categories: List data categories for which data is defined
+            data_category_column: Name of the data category column
+        """
+        self.data_file_path = data_file_path
+        self.metadata_file_path = metadata_file_path
+        self.mcf_file_path = mcf_file_path
+        self.csv_file_path = csv_file_path
+        self.tmcf_file_path = tmcf_file_path
+        self.existing_stat_var = existing_stat_var
+        self.census_year = census_year
+        self.dataset_name = dataset_name
+        self.raw_df = None
+        self.stat_var_index = {}
+        self.census_columns = []
+        self.data_categories = data_categories
+        self.data_category_column = data_category_column
+
+    def _format_data(self):
+
+        def stat_var_index_key(row, data_category_column=None):
+            key = "{0}_{1}".format(row["columnName"], row["TRU"])
+            if data_category_column:
+                key = key + "_" + row[data_category_column]
+            return key
+
+        #Generate Census locationid
```

Comment:
General comment style nit everywhere, have a single space between '#' and first character?

Reply:
Done.